ui/helpers/utils/util.js

```diff
@@ -631,3 +639,42 @@ export const getNetworkNameFromProviderType = (providerName) => {
 export const isAbleToExportAccount = (keyringType = '') => {
   return !keyringType.includes('Hardware') && !keyringType.includes('Snap');
 };
+
+/**
+ * Checks if a tokenId in Hex or decimal format already exists in an object.
+ *
+ * @param {string} address - collection address.
+ * @param {string} tokenId - tokenId to search for
+ * @param {*} obj - object to look into
+ * @returns {boolean} `false` if tokenId does not already exist.
+ */
+export const checkTokenIdExists = (address, tokenId, obj) => {
+  // check if input tokenId is hexadecimal
+  // If it is convert to decimal and compare with existing tokens
+  // if it is decimal convert to hexadecimal and compare
+  // if it is neither leave as it is
+  const isHex = isStrictHexString(tokenId);
```

Comment:
`isHex` will only be true if `tokenId` is `0x` prefixed

`!isNaN(tokenId)` will be true if `tokenId` can be cast to a number

is it possible for `tokenId` to be in hex, but not have a `0x` prefix? If so, it is possible that `tokenId` could be a hex number with only numeric digits, and we would be "converting" it to hex incorrectly below

For example, if tokenId is `'10'`, but that is hex and so equivalent to the decimal 16, but `!isNaN(tokenId)` would be true so we would convert it to `0xa` and so `isEqualCaseInsensitive(nft.tokenId, convertedTokenId.toString())` could return true if there is another `nft` with the `tokenId` `0xa`, even though we should be looking for an nft with the tokenId `'0x10'` or just `'10'` or the decimal number `10`

If we know that every `tokenId` passed to this function is either a `0x` prefixed hex number OR a decimal number, and there are no non-prefixed hex numbers, then I think this concern can be ignored.

Reply:
Although, this raises a question of why are there different types of token ids? Can we fix something somewhere else in the stack so that we know that all token ids are in hex?

Reply:
Nice point; in that scenario; if a user imports tokenId "0xa", and then tries to import tokenId "10" rather than "0x10" it will say that "the nft has already been imported".
But also if the user first adds tokenId (10 : decimal) and then tries to import again "10" but this time he meant "0x10", the  UI would also say that "NFT has already been imported".

Hmmm;  i guess the issue here is that the user can enter a "10" and we cannot tell if he meant that as a hex or a decimal.
The UI does not enforce adding a 0x and i do not think it should.

The reported  issue was when:
1- import nft with tokenId decimal
2- import same nft but with the equivalent strict hex representation
That alone could be solved by this version of the fct
```
export const checkTokenIdExists = (address, tokenId, obj) => {
  // check if input tokenId is hexadecimal
  // If it is convert to decimal and compare with existing tokens
  // if it is decimal convert to hexadecimal and compare
  // if it is neither leave as it is
  const isHex = isStrictHexString(tokenId);
  let convertedTokenId;
  if (isHex) {
    // Convert to decimal
    convertedTokenId = hexToDecimal(tokenId);
    // eslint-disable-next-line no-negated-condition
  }

  if (obj[address]) {
    const value = obj[address];
    return lodash.some(value.nfts, (nft) => {
      return (
        nft.address === address &&
        (isEqualCaseInsensitive(nft.tokenId, tokenId) ||
          isEqualCaseInsensitive(nft.tokenId, convertedTokenId.toString()))
      );
    });
  }
  return false;
};
```
By doing that  ☝️  we only avoid having duplicate nfts in one scenario (where user imports tokenId as decimal first)
Which i think is the common scenario.

But; if the user imports the tokenId in a hex format first, and then tries to import as a decimal then he will have a duplicate. Thinking about it, usually when dapps prompt the user to add an nft; they are always in decimal formats; so I'm not sure how common it would be for a user to import a hex format first.

But with that version of the function ☝️ , in the scenario that you mentioned, when the user first imports tokenId "0xa" and then imports tokenId "10" he will end up seeing a duplicate of the nft (tokenId 10) rather than having the UI say " NFT has already been imported"(committed version of the fact).

So i think that, with the current UI, either way we could end up with:
The user seeing error "NFT has already been imported" because he did not put the strict hex format
OR
The User seeing duplicate nfts when he imports hex format of tokenId first

Sorry if that was too long; there could be a simpler way that i have missed @danjm  🙏

Reply:
Thanks for the explanation. I think it makese sense to make the change you suggest for the common scenario.